### PR TITLE
regenerate the i18n files for the stocks example

### PR DESCRIPTION
**Pull request by devoncarew**
Not for landing:

This is an attempt to regenerate the i18n files from `stocks`; it strips all the localized strings out of the dart files however. @eseidelGoogle might have worked on this initially, and @alan-knight will have more context on how to generate dart files from `arb` ones.

### CODE REVIEW

File: examples/stocks/lib/i18n/regenerate.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+To rebuild the i18n files:
+
+pub run intl:generate_from_arb \
```
Comment (Hixie) [inline examples/stocks/lib/i18n/regenerate.md]: Put `\`\`\`` around this block

**Comment by eseidelGoogle** [comment]
@jason-simmons was actually the author of our original intl support. :)

**Comment by alan-knight** [comment]
Does it emit anything when it runs? It seems like it's not parsing the contents of the ARB files correctly. There were some changes to ARB parsing after looking at what Google Translation Toolkit was actually emitting.

**Comment by devoncarew** [comment]
oh:

```
No @@locale or _locale field found in stocks_en, assuming 'en' based on the file name.
No @@locale or _locale field found in stocks_es, assuming 'es' based on the file name.
```

So it looks like it doesn't like those files.

Can you take a look at them (https://github.com/flutter/flutter/tree/master/examples/stocks/lib/i18n) and see if anything stands out?

Also, it would be nice if the arb -> dart generator took a few less options, like if you could just point them at the arb files and it would infer most things from there.

**Comment by alan-knight** [comment]
That doesn't mean it doesn't like the ARB files, just that it's going by the file name rather than the internal metadata. So it is inferring the right thing.

The problem is, um, "obvious" in retrospect. The input to generate_from_arb is the ARB files and the Dart files which contain the original calls to be localized. That is, the argument should be lib/*.dart (plus subdirectories if there were any), not lib/i18n/*.dart. It parses the original files, looking for anything that defines an Intl.message that needs translation. When it's parsing its own output there's nothing requiring translation, so it generates empty translation libraries. But it's an easy mistake to make and a hard one to figure out, so there should be something that detects that case.

**Comment by devoncarew** [comment]
@alan-knight, thanks! The generated files look good now; updated this PR.

**Comment by devoncarew** [comment]
(updated an analysis regex)

File: examples/stocks/lib/i18n/regenerate.md (hunk 1 above)
Comment (devoncarew) [inline examples/stocks/lib/i18n/regenerate.md]: done

**Comment by eseidelGoogle** [comment]
Is the goal here to just update our i18n dependency? or are these new ones more analyzer compliant?  Just missing context.

**Comment by devoncarew** [comment]
We've had some lint issues with these files. These newer versions of the files are slightly happier wrt linting. I also didn't know how to generate them in the first place, so this adds some instructions for how to do that.

**Comment by eseidelGoogle** [comment]
lgtm.  We'll probably want to teach the flutter tool how to do this at some point.

**Pull request merged**